pipeline.py

```diff
@@ -72,31 +88,82 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+            article.save_as(*cleaned_tokens, kind=ArtifactType.cleaned)
+            article.save_as(*single_tagged_tokens, kind=ArtifactType.single_tagged)
+            print(*cleaned_tokens)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        mystem = Mystem()
+        analyzed_text = mystem.analyze(raw_text)
+        tokens = []
+        for analyzed_word in analyzed_text:
+            if not analyzed_word.get('analysis'):
+                continue
+            morphological_token = MorphologicalToken(original_word=analyzed_word.get('text'))
+            morphological_token.normalized_form = analyzed_word.get('analysis')[0].get('lex')
+            morphological_token.mystem_tags = analyzed_word.get('analysis')[0].get('gr')
+            tokens.append(morphological_token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+    if not path.exists():
+        raise FileNotFoundError
+    if not path.is_dir():
+        raise NotADirectoryError
+
+    all_article_ids = []
+    for file in path.glob("*.txt"):
+        name_pattern = re.match(r'\d+_(raw.txt)', file.name)
```

Comment:
you could already extract ID instead of creating several patterns, and even worse - in a loop